pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -117,3 +121,11 @@ func startServicesInformer(ctx ControllerContext) error {
 
 	return nil
 }
+
+func startEndpointsInformer(ctx ControllerContext) error {
+	// Just start the shared informer, the autodiscovery
+	// components will access it when needed.
+	go ctx.InformerFactory.Core().V1().Endpoints().Informer().Run(ctx.StopCh)
```

Comment:
do you think we need to add `sync.OnlyOnce` here?

Reply:
I don't think so, the func is private and used to populate the `controllerCatalog` only (same as `startServicesInformer` `startAutoscalersController` and `startMetadataController`. 
it's called by the public func `StartControllers` which is called once at the cluster agent entrypoint https://github.com/DataDog/datadog-agent/blob/master/cmd/cluster-agent/app/app.go#L203